tests/Integration/inc/Engine/CriticalPath/APIClient/sendGenerationRequest.php

```diff
@@ -0,0 +1,75 @@
+<?php
+namespace WP_Rocket\Tests\Integration\inc\Engine\CriticalPath\APIClient;
+
+use WP_Rocket\Engine\CriticalPath\APIClient;
+use WP_Rocket\Tests\Integration\FilesystemTestCase;
+use Brain\Monkey\Functions;
+use WP_Error;
+
+/**
+ * @covers \WP_Rocket\Engine\CriticalPath\APIClient::send_generation_request
+ * @group CriticalPath
+ * @group  vfs
+ */
+class Test_SendGenerationRequest extends FilesystemTestCase {
+
+	protected $path_to_test_data = '/inc/Engine/CriticalPath/APIClient/sendGenerationRequest.php';
+
+	/**
+	 * @dataProvider nonMultisiteTestData
+	 */
+	public function testShouldDoExpected( $config, $expected ) {
+		$item_url      = isset( $config['item_url'] ) ? $config['item_url'] : '';
+		$response_code = ! isset( $config['response_data']['code'] ) ? 200 : $config['response_data']['code'];
+		$response_body = ! isset( $config['response_data']['body'] ) ? '' : $config['response_data']['body'];
+
+		Functions\expect( 'wp_remote_post' )
+			->atMost()
+			->times( 1 )
+			->with(
+				'https://cpcss.wp-rocket.me/api/job/',
+				[
+					'body' => [
+						'url' => $item_url,
+					],
+				]
+			)
+			->andReturn( 'postRequest' );
+
+		Functions\expect( 'wp_remote_retrieve_response_code' )
+			->atMost()
+			->times( 1 )
+			->with( 'postRequest' )
+			->andReturn( $response_code );
+
+		Functions\expect( 'wp_remote_retrieve_body' )
+			->atMost()
+			->times( 1 )
+			->with( 'postRequest' )
+			->andReturn( $response_body );
```

Comment:
Instead of `atMost, be specific to validate the expected behavior with these mocks. [Use this method as a guide.](https://github.com/wp-media/wp-rocket/blob/feature/cpcss-refactor/tests/Unit/inc/Engine/CriticalPath/APIClient/getJobDetails.php#L41)